### ISSUE TITLE
Error when setting up boundary conditions

### ISSUE THREAD

**Issue by weiyilan** [SynxFlow/SynxFlow#3]
I put a new dem and a landcover and rainmask just one zone, it reads dem, landcover and rainmask fine. But stops when setting boundary conditions. Attached is the dem, landcover and 
![1](https://github.com/SynxFlow/SynxFlow/assets/43534645/7d4f2547-d1f7-4203-8395-f22365a4245a)
![2](https://github.com/SynxFlow/SynxFlow/assets/43534645/214c04ac-6847-4265-89b1-6f264408c066)
![3](https://github.com/SynxFlow/SynxFlow/assets/43534645/c33e4123-d615-4899-9ef9-173737b9668f)
error message.

**Comment by mingxiaodong** [comment]
Could you post your code to setup boundary condition?

**Comment by weiyilan** [comment]
# set boundary conditions
import numpy as np
box_upstream = np.array([[1994681, 405040],  # bottom left
                         [1995101, 404552]]) # upper right
box_downstream = np.array([[2069000, 465653], # upper left
                           [2072000, 454500]]) # bottom right
discharge_values = np.array([[0, 1780], # first column: time - s; second colum: discharge - m3/s
                            [7200,1780]])

bound_list = [
            {'polyPoints': box_upstream,
             'type': 'open',
             'hU': discharge_values},
            {'polyPoints': box_downstream,
             'type': 'open',
             'h': np.array([[0, 5],
                            [7200,5]])}] # we fix the downstream depth as 12.5 m
case_input.set_boundary_condition(boundary_list=bound_list)
case_input.domain_show() # show domain map

**Comment by mingxiaodong** [comment]
When you define a boundary box consisting of two points, the box should cover a segment of the dem outline. Otherwise, it can't find any boundary cells. In your case, its seems like your defined boxes are either all inside or outside the dem domain. But I am not quite sure according to your code as I don't have your DEM data. I suggest you to have a check at the box coordinates or plot the box points on the dem map using the following code:

```
fig, ax = DEM.mapshow()
ax.plot(box_upstream[:, 0], box_upstream[:, 1], 'r') 
ax.plot(box_downstream[:, 0], box_downstream[:, 1], 'y')
```
You should be able to see the diagonals of the two boxes. Are they across the dem outline?

**Comment by weiyilan** [comment]
![Figure_1](https://github.com/SynxFlow/SynxFlow/assets/43534645/d060706d-0d04-4900-a342-28ca6012e7af)
It intersects the outline. Bye the way, when you say "box", I am really confused. With two vertices, it only creates a line, why a box? From the plot, those are lines. And if are lines, then lines intersect each other can only make several points.

**Comment by weiyilan** [comment]
![Figure_1](https://github.com/SynxFlow/SynxFlow/assets/43534645/5c2285dc-d214-46c9-82f1-42ef92df1c8f)
Oh. My mistake. I thought you mean only outflow bc needs to be intersecting the outline. It turned out be to both inflow and outflow bc needs to be intersecting the outline. I thought it is like HECRAS. After I change both bc to be intersecting the outline. It works. Thanks a lot for your help!

**Comment by mingxiaodong** [comment]
Yes, all the boundary boxes need to intersect with the dem outline. 

Each of the two vetices has x and y coordinates that can together define x_min, x_max, y_min, and y_max of **a box parallel with the x and y axes**. That's why we only need two points to define a box. The lines you have drawn in the map are diagonal lines of the boxes. If a line made you confusing, you can draw a box based on the two vetices and see whether the box covers a segment of the dem outline.  For example,
```
fig, ax = DEM.mapshow()
# define x_min, x_max, y_min, and y_max
box_coords = np.array([[x_min, ymin], [x_max, y_min], [x_max, y_max], [x_min, y_max], [x_min, ymin]])
ax.plot(box_coords[:, 0], box_coords[:, 1], 'r') 
```
On your latest map, the box indicated by the yellow line can clearly cover a segment of the dem outline. The red line might only cover a tiny bit of the outline. You can always adjust the location of the two vetices, ideally one inside the dem domain and one outside to select the boundary cells.
